### ISSUE TITLE
[wasm] Test run crashing with `exiting due to exception: 613560632` from V8, and browser

### ISSUE THREAD

**Issue by radical** [dotnet/runtime#74303]
[Build](https://dev.azure.com/dnceng/public/_build/results?buildId=1951159&view=results), and [log](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-1fae416268b942518b/normal-System.Collections.Tests/1/console.ebd1aa4e.log?helixlogtype=result):

```
Discovering: System.Collections.Tests.dll (method display = ClassAndMethod, method display options = None)
Discovered:  System.Collections.Tests.dll (found 5554 of 7453 test cases)
Using random seed for test cases: 1610090969
Using random seed for collections: 1610090969
Starting:    System.Collections.Tests.dll
613560632
WASM EXIT 1
613560632
exiting due to exception: 613560632
Process v8 exited with 1
```

This has been seen only once, yet.

**Comment by pavelsavara** [comment]
Is this duplicate of https://github.com/dotnet/runtime/issues/74302 ?

**Comment by radical** [comment]
Another [instance](https://dev.azure.com/dnceng/public/_build/results?buildId=1971504&view=results)([log](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-364163df84ae48ea86/normal-System.Collections.Tests/1/console.bd796e68.log?helixlogtype=result)):

```
10:24:34] info: console.debug: mono_wasm_runtime_ready fe00e07a-5519-4dfe-b35a-f867dbaf2e28
[10:24:34] info: console.info: Initializing.....
[10:24:34] info: Discovering: System.Collections.Tests.dll (method display = ClassAndMethod, method display options = None)
[10:24:38] info: Discovered:  System.Collections.Tests.dll (found 5554 of 7453 test cases)
[10:24:38] info: Using random seed for test cases: 891169863
[10:24:38] info: Using random seed for collections: 891169863
[10:24:38] info: Starting:    System.Collections.Tests.dll
[10:26:00] info: WASM EXIT 1
[10:26:00] info: 221033864
[10:26:00] info: 221033864
[10:26:00] info: exiting due to exception: 221033864
[10:26:00] info: Process v8 exited with 1
```

**Comment by radical** [comment]
This is being hit on Chrome/windows too, with rolling builds. It was seen in 6 out of 7 rolling builds for `System.Private.Xml.Tests`/AOT.

[1](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-b4bfbbfc494b41eb96/WasmTestOnBrowser-System.Private.Xml.Tests/1/console.bac91a10.log?helixlogtype=result), [2](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-5d9d19187fdd4eb49d/WasmTestOnBrowser-System.Private.Xml.Tests/1/console.8036c146.log?helixlogtype=result), and [3](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-61a326dc2b8f4d0eb4/WasmTestOnBrowser-System.Private.Xml.Tests/1/console.efae9b84.log?helixlogtype=result).

**Comment by radical** [comment]
AOT with chrome:

```
  info: Using random seed for collections: 1903691008
  info: Starting:    System.Private.Xml.Tests.dll
  fail: Error: 149575608
            at mono_wasm_stringify_as_error_with_stack (http://127.0.0.1:35093/dotnet.js:799:22)
            at console.error (http://127.0.0.1:35093/dotnet.js:866:35)
            at logErrorOnExit (http://127.0.0.1:35093/dotnet.js:1079:29)
            at set_exit_code_and_quit_now (http://127.0.0.1:35093/dotnet.js:1049:9)
            at mono_exit (http://127.0.0.1:35093/dotnet.js:1010:13)
            at run (http://127.0.0.1:35093/test-main.js:335:21)
  fail: Error: 149575608
            at mono_wasm_stringify_as_error_with_stack (http://127.0.0.1:35093/dotnet.js:799:22)
            at console.error (http://127.0.0.1:35093/dotnet.js:866:35)
            at logErrorOnExit (http://127.0.0.1:35093/dotnet.js:1079:29)
            at set_exit_code_and_quit_now (http://127.0.0.1:35093/dotnet.js:1049:9)
            at mono_exit (http://127.0.0.1:35093/dotnet.js:1010:13)
            at run (http://127.0.0.1:35093/test-main.js:342:9)
  fail: Error: 149575608
            at mono_wasm_stringify_as_error_with_stack (http://127.0.0.1:35093/dotnet.js:799:22)
            at console.error (http://127.0.0.1:35093/dotnet.js:866:35)
            at logErrorOnExit (http://127.0.0.1:35093/dotnet.js:1079:29)
            at set_exit_code_and_quit_now (http://127.0.0.1:35093/dotnet.js:1049:9)
            at mono_exit (http://127.0.0.1:35093/dotnet.js:1010:13)
            at fatal_handler (http://127.0.0.1:35093/dotnet.js:9610:21)
            at http://127.0.0.1:35093/dotnet.js:9619:78
  fail: Error: 149576856
            at mono_wasm_stringify_as_error_with_stack (http://127.0.0.1:35093/dotnet.js:799:22)
            at console.error (http://127.0.0.1:35093/dotnet.js:866:35)
            at logErrorOnExit (http://127.0.0.1:35093/dotnet.js:1079:29)
            at set_exit_code_and_quit_now (http://127.0.0.1:35093/dotnet.js:1049:9)
            at mono_exit (http://127.0.0.1:35093/dotnet.js:1010:13)
            at fatal_handler (http://127.0.0.1:35093/dotnet.js:9610:21)
            at http://127.0.0.1:35093/dotnet.js:9620:65
```

**Comment by radical** [comment]
> This is being hit on Chrome/windows too, with rolling builds. It was seen in 6 out of 7 rolling builds for `System.Private.Xml.Tests`/AOT. Three of those examples: [1](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-b4bfbbfc494b41eb96/WasmTestOnBrowser-System.Private.Xml.Tests/1/console.bac91a10.log?helixlogtype=result), [2](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-5d9d19187fdd4eb49d/WasmTestOnBrowser-System.Private.Xml.Tests/1/console.8036c146.log?helixlogtype=result), and [3](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-61a326dc2b8f4d0eb4/WasmTestOnBrowser-System.Private.Xml.Tests/1/console.efae9b84.log?helixlogtype=result).

This is happening on every rolling build now. We are missing coverage for `System.Private.Xml.Tests`.

**Comment by radical** [comment]
Hit on linux/AOT on a rolling build for *7.0* branch's [log](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-release-70-210beef37f9545bfa1/normal-System.Collections.Tests/1/console.25563072.log?helixlogtype=result), [build](https://dev.azure.com/dnceng-public/public/_build/results?buildId=71652&view=results):

```
[06:40:04] info: Discovered:  System.Collections.Tests.dll (found 5554 of 7453 test cases)
[06:40:04] info: Using random seed for test cases: 1124391650
[06:40:04] info: Using random seed for collections: 1124391650
[06:40:04] info: Starting:    System.Collections.Tests.dll
[06:41:01] info: WASM EXIT 1
[06:41:01] info: 1349757600
[06:41:01] info: 1349757600
[06:41:01] info: exiting due to exception: 1349757600
[06:41:01] info: Process v8 exited with 1
```

**Comment by pavelsavara** [comment]
@vargaz @radical 
Is this still happening ? Or was it fixed by  https://github.com/dotnet/runtime/pull/76743

**Comment by vargaz** [comment]
It should be fixed.
